File: commonrv/src/main/java/com/even/commonrv/utils/GlideUtil.java
package com.even.commonrv.utils;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.RequestManager;
import com.bumptech.glide.request.RequestListener;
import com.bumptech.glide.request.RequestOptions;

import java.io.File;

/**
 * @author by Even on 2018/11/1
 * Emial: emailtopan@163.com
 * 显示相关工具类
 */
public class GlideUtil {

    /**
     * 加载网络
     */
    public static void loadNet(ImageView target, String url, RequestOptions options, RequestListener<Drawable> listener) {
        getRequestManager(target.getContext()).load(url).listener(listener).apply(options).into(target);
    }

    /**
     * 加载网络
     */
    public static void loadNet(ImageView target, String url, RequestOptions options) {
        getRequestManager(target.getContext()).load(url).apply(options).into(target);
    }

    /**
     * 加载网络
     */
    public static void loadNet(ImageView target, String url) {
        getRequestManager(target.getContext()).load(url).into(target);
    }

    /**
     * 加载本地资源
     */
    public static void loadResource(ImageView target, int resId) {
        getRequestManager(target.getContext()).load(resId).into(target);
    }

    /**
     * 加载文件,带加载监听
     */
    public static void loadFile(ImageView target, File file, RequestOptions options, RequestListener<Drawable> listener) {
        getRequestManager(target.getContext()).load(file).listener(listener).apply(options).into(target);
    }

    /**
     * 加载文件
     */
    public static void loadFile(ImageView target, File file, RequestOptions options) {
        getRequestManager(target.getContext()).load(file).apply(options).into(target);
    }

    /**
     * 加载本地文件
     */
    public static void loadFile(ImageView target, File file) {
        getRequestManager(target.getContext()).load(file).into(target);
    }

    /**
     * 清除内存缓存
     */
    public static void clearMemoryCache(Context context) {
        Glide.get(context).clearMemory();
    }

    /**
     * 清除磁盘缓存
     */
    public static void clearDiskCache(Context context) {
        Glide.get(context).clearDiskCache();
    }

    private static RequestManager getRequestManager(Context context) {
        return Glide.with(context);
    }


}

File: commonrv/src/main/java/com/even/commonrv/utils/GlideUtil.kt
package com.even.commonrv.utils

import android.app.Activity
import android.content.Context
import android.graphics.drawable.Drawable
import android.widget.ImageView
import com.bumptech.glide.Glide
import com.bumptech.glide.RequestManager
import com.bumptech.glide.request.RequestListener
import com.bumptech.glide.request.RequestOptions
import java.io.File

/**
 * @author by Even on 2018/11/1
 * Email: emailtopan@163.com
 * 图片加载相关工具类
 */
object GlideUtil {
    /**
     * 加载网络
     */
    fun loadNet(
        target: ImageView,
        url: String,
        options: RequestOptions,
        listener: RequestListener<Drawable?>
    ) {
        if (checkIsDestroyedActivity(target.context as Activity)) {
            return
        }
        getRequestManager(target.context).load(url).listener(listener).apply(options)
            .into(target)
    }

    /**
     * 加载网络
     */
    fun loadNet(
        target: ImageView,
        url: String,
        options: RequestOptions
    ) {
        if (checkIsDestroyedActivity(target.context as Activity)) {
            return
        }
        getRequestManager(target.context).load(url).apply(options).into(target)
    }

    /**
     * 加载网络
     */
    fun loadNet(target: ImageView, url: String?) {
        if (checkIsDestroyedActivity(target.context as Activity)) {
            return
        }
        getRequestManager(target.context).load(url).into(target)
    }

    /**
     * 加载本地资源
     */
    fun loadResource(target: ImageView, resId: Int) {
        if (checkIsDestroyedActivity(target.context as Activity)) {
            return
        }
        getRequestManager(target.context).load(resId).into(target)
    }

    /**
     * 加载文件,带加载监听
     */
    fun loadFile(
        target: ImageView,
        file: File,
        options: RequestOptions,
        listener: RequestListener<Drawable?>
    ) {
        if (checkIsDestroyedActivity(target.context as Activity)) {
            return
        }
        getRequestManager(target.context).load(file).listener(listener)
            .apply(options).into(target)
    }

    /**
     * 加载文件
     */
    fun loadFile(
        target: ImageView,
        file: File,
        options: RequestOptions
    ) {
        if (checkIsDestroyedActivity(target.context as Activity)) {
            return
        }
        getRequestManager(target.context).load(file).apply(options).into(target)
    }

    /**
     * 加载本地文件
     */
    fun loadFile(target: ImageView, file: File) {
        if (checkIsDestroyedActivity(target.context as Activity)) {
            return
        }
        getRequestManager(target.context).load(file).into(target)
    }

    /**
     * 清除内存缓存
     */
    fun clearMemoryCache(context: Context) {
        Glide.get(context).clearMemory()
    }

    /**
     * 清除磁盘缓存
     */
    fun clearDiskCache(context: Context) {
        Glide.get(context).clearDiskCache()
    }

    private fun getRequestManager(context: Context): RequestManager {
        return Glide.with(context)
    }

    /**判断Activity是否已回收，防止You cannot start a load for a destroyed activity错误*/
    fun checkIsDestroyedActivity(activity: Activity): Boolean {
        return activity.isFinishing
    }
}